test/components/views/settings/AddPrivilegedUsers-test.tsx

```diff
@@ -0,0 +1,143 @@
+import userEvent from "@testing-library/user-event";
+import { mocked } from "jest-mock";
+import { RoomMember } from "matrix-js-sdk/src/matrix";
+
+import {
+    getMockClientWithEventEmitter,
+    makeRoomWithStateEvents,
+} from "../../../test-utils";
+import MatrixClientContext from '../../../../src/contexts/MatrixClientContext';
+import {
+    AddPrivilegedUsers,
+    getUserIdsFromCompletions, hasLowerOrEqualLevelThanDefaultLevel,
+} from "../../../../src/components/views/settings/AddPrivilegedUsers";
+import UserProvider from "../../../../src/autocomplete/UserProvider";
+import { ICompletion } from "../../../../src/autocomplete/Autocompleter";
+
+jest.mock('../../../../src/autocomplete/UserProvider');
+
+const completions: ICompletion[] = [
+    { type: 'user', completion: 'user_1', completionId: '@user_1:host.local', range: { start: 1, end: 1 } },
+    { type: 'user', completion: 'user_2', completionId: '@user_2:host.local', range: { start: 1, end: 1 } },
+    { type: 'user', completion: 'user_without_completion_id', range: { start: 1, end: 1 } },
+];
+
+describe('<AddPrivilegedUsers />', () => {
+    const provider = mocked(UserProvider, { shallow: true });
+    provider.prototype.getCompletions.mockResolvedValue(completions);
+
+    const mockClient = getMockClientWithEventEmitter({
+        // `makeRoomWithStateEvents` only work's if `getRoom` is present.
+        getRoom: jest.fn(),
+        setPowerLevel: jest.fn(),
+    });
+
+    const room = makeRoomWithStateEvents([], { roomId: 'room_id', mockClient: mockClient });
+    room.getMember = (userId: string) => {
+        const member = new RoomMember('room_id', userId);
+        member.powerLevel = 0;
+
+        return member;
+    };
+
+    const getComponent = () =>
+        <MatrixClientContext.Provider value={mockClient}>
+            <AddPrivilegedUsers
+                room={room}
+                defaultUserLevel={0}
+            />
+        </MatrixClientContext.Provider>;
+
+    it('checks whether form submit works as intended', async () => {
+        const { getByTestId, queryAllByTestId } = render(getComponent());
+
+        // Verify that the submit button is disabled initially.
+        const submitButton = getByTestId('add-privileged-users-submit-button');
+        expect(submitButton).toBeDisabled();
+
+        // Find some suggestions and select them.
+        const autocompleteInput = getByTestId('autocomplete-input');
+
+        act(() => {
+            fireEvent.focus(autocompleteInput);
+            fireEvent.change(autocompleteInput, { target: { value: 'u' } });
+        });
+
+        await waitFor(() => expect(provider.mock.instances[0].getCompletions).toHaveBeenCalledTimes(1));
+        const matchOne = getByTestId('autocomplete-suggestion-item-@user_1:host.local');
+        const matchTwo = getByTestId('autocomplete-suggestion-item-@user_2:host.local');
+
+        act(() => {
+            fireEvent.mouseDown(matchOne);
+        });
+
+        act(() => {
```

Comment:
Can this be merged to the previous `act`?

Reply:
Actually started with a merged `act` and divided it because it failed 🤔

Reply:
Ok, lets keep it for now